### PR TITLE
Implement `measuretype_taxonomies` API

### PR DESCRIPTION
**Pull request by parndt**
Implements #26

We need a many to many join table to create a relationship between
`measuretypes`, introduced for #25, and `taxonomies`, from impact.

##### Permissions by role

- public: none
- registered, no role: none
- analyst: R
- manager: R
- admin: R

### CODE REVIEW

**Comment by parndt** [comment]
This contains code from #61 so needs that merged first

File: db/schema.rb [hunk 1]
```diff
@@ -191,10 +183,27 @@
...
 
+  create_table "measuretypes", force: :cascade do |t|
+    t.string "title", null: false
+    t.boolean "has_target", default: true
+    t.boolean "has_parent", default: true
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+  end
+
+  create_table "measuretypes_taxonomies", force: :cascade do |t|
```
Comment (tmfrnz) [inline db/schema.rb:198]: table should be called measuretype_taxonomies

File: db/migrate/20211103075455_create_measuretypes_taxonomies.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateMeasuretypesTaxonomies < ActiveRecord::Migration[6.1]
+  def change
+    create_table :measuretypes_taxonomies do |t|
```
Comment (tmfrnz) [inline db/migrate/20211103075455_create_measuretypes_taxonomies.rb:3]: table should be called measuretype_taxonomies

**Review by tmfrnz — APPROVED** [review]
may want to rename that join table (measuretype_taxonomies) eventually but happy to go ahead and merge this regardless

File: db/migrate/20211103075455_create_measuretypes_taxonomies.rb (hunk 2 above)
Comment (parndt) [reply db/migrate/20211103075455_create_measuretypes_taxonomies.rb:3]: Not in Rails conventions it shouldn't. Many to many tables have plurals on both sides

Comment (parndt) [reply db/migrate/20211103075455_create_measuretypes_taxonomies.rb:3]: Can override the convention if you really want to 😄

**Pull request merged**